include/boost/math/fft/fftw_backend.hpp

```diff
@@ -624,6 +624,11 @@
       std::copy(in,in+size(),out);
       unpack_halfcomplex(out);
       execute(my_hc2r_plan,my_hc2r_unaligned_plan,out,out);
+      
+      const std::size_t N = size();
+      const real_value_type inv_N = real_value_type{1.0}/N;
+      for(unsigned int i=0;i<N;++i)
+        out[i] *= inv_N;
```

Comment:
when transforming halfcomplex-to-real, both GSL and FFTW, return an array which is a 1/N factor of the original real array. This could be in tune with the fact that the reverse FFT is the inverse of the forward FFT up to a 1/N factor. But the actual implementation of the halfcomplex-to-real (at least in BSL) is the reverse graph of the real-to-halfcomplex and the 1/N factor does not appear. I cannot see a good reason why the halfcomplex-to-real cannot be the inverse linear map of the real-to-halfcomplex.

Reply:
@Lagrang3 yes, this is a good idea. Better to make sure that all backends produce the same kind of output. To avoid any kind of frustration for the users who have a nice working code, and only decide to switch to a different backend.

Having to put an if somewhere in the user code to multiply by 1/N, depending on the type of backend, would not be good.